kfs-core/src/main/java/edu/arizona/kfs/gl/batch/service/impl/BudgetAdjustmentCashTransferServiceImpl.java

```diff
@@ -0,0 +1,596 @@
+                         throw new RuntimeException("Error creating cash transfer done file", e);
+                     }
+            	 
+                 }                
+                        
+             }
+           
+             //delete all budget adjustment transactions  
+             budgetAdjustmentTransactionDao.deleteAllBudgetAdjustmentTransactions();
+             OUTPUT_GLE_FILE_ps.close();
+             reportWriterService.writeStatisticLine("GLBA TRANSACTIONS LOADED        (GL_BUDGET_ADJUST_TRN_T) %,9d", reportBudgetAdjustDocLoaded);
+             reportWriterService.writeStatisticLine("CASH TRANSFER TRANSACTIONS GENERATED                     %,9d", reportOriginEntryGenerated);             
+
+         }
+         catch (FileNotFoundException e) {
+             throw new RuntimeException("generateBudgetAdjustmentCashTransferTransactions Stopped: " + e.getMessage(), e);
+         }
+    }
+    
+    /**
+     * Returns a list of unique document numbers which are needed so that we can
+     * group budget adjustment transactions by budget system document number in
+     * order to determine if the transactions, as a group, cross income streams.
+      *
+     * @return docNumberList list of unique budget adjustment document numbers      
+     */
+    protected List<String> retrieveUniqueDocumentNumbersList() {     	
+    	 Iterator budgetAdjustmentTransactions;
+    	 List<String> docNumberList = new ArrayList<String>();
+    	 
+    	 try {
+             budgetAdjustmentTransactions = budgetAdjustmentTransactionDao.getAllBudgetAdjustmentTransactions();
+         }
+         catch (RuntimeException re) {
+             LOG.error("generateBudgetAdjustmentCashTransferTransactions Stopped: " + re.getMessage());
+             throw new RuntimeException("generateBudgetAdjustmentCashTransferTransactions Stopped: " + re.getMessage(), re);
+         }
+         
+         while (budgetAdjustmentTransactions.hasNext()) {
+             BudgetAdjustmentTransaction ba = new BudgetAdjustmentTransaction();
+             try {
+                 ba = (BudgetAdjustmentTransaction) budgetAdjustmentTransactions.next();
+                                  
+                 if (!docNumberList.contains(ba.getDocumentNumber())) {
+                	 docNumberList.add(ba.getDocumentNumber());                     
+                 }                      
+             }
+             catch (RuntimeException re) {
+                 LOG.error("generateBudgetAdjustmentCashTransferTransactions Stopped: " + re.getMessage());
+                 throw new RuntimeException("generateBudgetAdjustmentCashTransferTransactions Stopped: " + re.getMessage(), re);
+             }
+             catch (Exception e) {
+                 List errorList = new ArrayList();
+                 errorList.add(new Message(e.toString() + " occurred for this record.", Message.TYPE_FATAL));
+                 reportWriterService.writeError(ba, errorList);
```

Comment:
Inconsistent Exception handling: if we get a RuntimeException, we don't use the reportWriterService to writeError but we rethrow it. If we catch any kind of other exception, we just log it with the reportWriterService.writeError and we continue??